pxr/base/tf/hash.h

```diff
@@ -94,6 +97,33 @@ TfHashAppend(HashState &h, std::vector<T> const &vec)
     h.AppendContiguous(vec.data(), vec.size());
 }
 
+// Support std::set
+template <class HashState, class T>
+inline void
+TfHashAppend(HashState& h, std::set<T> const &elements)
```

Comment:
Why do we not support custom comparators for std::set but we do for std::map?

Reply:
No reason other than it became necessary for `std::map` in other downstream changes but not `std::set`.  I'm updating `std::set`.